### PR TITLE
WIP: Support deploying operator via OLM

### PR DESCRIPTION
**Pull request by dfarrell07**
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Comment by skitt** [comment]
LGTM so far...

**Comment by mangelajo** [comment]
Remember that we don't want to remove the non-olm behavior, we should use one or the other depending on what we find :)

File: pkg/subctl/operator/common/olmsubscription/utils.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		// But not on the master branch, and it fails verify:
+		// type "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1".SubscriptionStatus has no field or method Conditions
+		//for _, cond := range sub.Status.Conditions {
+		//	 TODO: I see "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1".SubscriptionCatalogSourcesUnhealthy here:
+		//	 https://github.com/operator-framework/operator-lifecycle-manager/blob/release-4.4/pkg/api/apis/operators/v1alpha1/subscription_types.go#L95
+		//	 But not on the master branch, and it fails verify:
+		//	 SubscriptionCatalogSourcesUnhealthy not declared by package v1alpha1
+		//	if cond.Type == olmv1alpha1.SubscriptionCatalogSourcesUnhealthy && cond.Status == v1.ConditionFalse {
+		//		return true, nil
+		//	}
```
Comment (dfarrell07) [inline pkg/subctl/operator/common/olmsubscription/utils.go:55]: @skitt (or others) any hints about what I'm doing wrong here (as desrbied in the inline TODOs)?

Comment (skitt) [reply pkg/subctl/operator/common/olmsubscription/utils.go:55]: The APIs were moved from `operator-framework/operator-lifecycle-manager` to `operator-framework/api`, and are present directly in `pkg/operators` there, unversioned (at least for `subscription_types`); that’s what the OLM clientset now uses.

Comment (dfarrell07) [reply pkg/subctl/operator/common/olmsubscription/utils.go]: Thanks @skitt, that helped! As I updated on the Automation WG call this week, I ended up realizing I needed to stick to the old 1.14.1 code due to the pinning going on elsewhere. The latest version is past this part and installing the Operator using OLM locally for me (although I'm seeing what seem to be timeouts here in CI).

**Comment by dfarrell07** [comment]
I think building on @SteveMattar's work we can get this working, so leaving it open.

**Comment by SteveMattar** [comment]
> I think building on @SteveMattar's work we can get this working, so leaving it open.

@dfarrell07 #724

**Pull request closed without merging**